### PR TITLE
chore: split the server tests into individual tests

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
This splits the individual `mysql`, `postgres`, and `memory` server tests into individual Go tests so that they can be run individually if you prefer. This is particularly helpful during development because then you can just run the `memory` tests for quick feedback.

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -37,35 +37,35 @@ func init() {
...
-		uri := testDatastore.GetConnectionURI()
-		ds, err := mysql.NewMySQLDatastore(uri)
-		require.NoError(t, err)
-		defer ds.Close()
+	test.RunAllTests(t, ds)
+}
 
-		test.RunAllTests(t, ds)
-	})
+func TestOpenFGAServer(t *testing.T) {
```
Comment (jon-whit) [inline server/server_test.go:65]: Don't think this is needed anymore. Let me double check that we're not referencing it externally somehow..

Comment (jon-whit) [reply server/server_test.go:65]: Yep, not needed. We only use `RunAllTests` externally.

**Review by craigpastro — DISMISSED** [review]
LGTM

File: server/server_test.go [hunk 2]
```diff
@@ -37,35 +37,29 @@ func init() {
...
 
-	t.Run("TestPostgresDatastore", func(t *testing.T) {
-		testDatastore := storagefixtures.RunDatastoreTestContainer(t, "postgres")
-
-		uri := testDatastore.GetConnectionURI()
-		ds, err := postgres.NewPostgresDatastore(uri)
-		require.NoError(t, err)
-		defer ds.Close()
+	uri := testDatastore.GetConnectionURI()
+	ds, err := postgres.NewPostgresDatastore(uri)
```
Comment (miparnisari) [inline server/server_test.go:44]:
```suggestion
	ds, err := postgres.NewPostgresDatastore(uri)
	defer ds.Close()
```

File: server/server_test.go [hunk 3]
```diff
@@ -37,35 +37,29 @@ func init() {
...
+	test.RunAllTests(t, ds)
+}
 
-	t.Run("TestMemoryDatastore", func(t *testing.T) {
-		ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
-		defer ds.Close()
-		test.RunAllTests(t, ds)
-	})
+func TestServerWithMemoryDatastore(t *testing.T) {
+	ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
```
Comment (miparnisari) [inline server/server_test.go:51]:
```suggestion
	ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
	defer ds.Close()
```

File: server/server_test.go [hunk 4]
```diff
@@ -37,35 +37,29 @@ func init() {
...
-		testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
+func TestServerWithMySQLDatastore(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
 
-		uri := testDatastore.GetConnectionURI()
-		ds, err := mysql.NewMySQLDatastore(uri)
-		require.NoError(t, err)
-		defer ds.Close()
+	uri := testDatastore.GetConnectionURI()
+	ds, err := mysql.NewMySQLDatastore(uri)
```
Comment (miparnisari) [inline server/server_test.go:59]:
```suggestion
	ds, err := mysql.NewMySQLDatastore(uri)
	defer ds.Close()
```

**Review by craigpastro — APPROVED** [review]
LGTM

**Pull request merged**